server/src/main/java/org/jboss/as/server/deployment/Attachments.java

```diff
@@ -125,14 +124,12 @@ public final class Attachments {
      * <p/>
      * <strong>Note: This is only meant for use within the server kernel.</strong>
      *
-     * @deprecated this will either be changed incompatibly (to provide a string) or removed altogether in the next WildFly Core major.
      */
-    @Deprecated(forRemoval = true)
-    public static final AttachmentKey<AttachmentList<ModuleIdentifier>> CLASS_PATH_ENTRIES = AttachmentKey.createList(ModuleIdentifier.class);
+    public static final AttachmentKey<AttachmentList<String>> CLASS_PATH_MODULES = AttachmentKey.createList(String.class);
```

Comment:
I've realized this doesn't need to be in this class, so I'll move it to the only package where it's used.